lib/rules/semi-spacing.js

```diff
@@ -104,22 +104,52 @@ module.exports = function(context) {
 
             if (hasLeadingSpace(token)) {
                 if (!requireSpaceBefore) {
-                    context.report(node, location, "Unexpected whitespace before semicolon.");
+                    context.report({
```

Comment:
I am sorry but based on the current change, i think it was better off before.
Lets just leave it as you had before.

Thanks